### PR TITLE
Search by regiment name

### PR DESCRIPTION
**Pull request by DavidColemanUK**
Awaiting style and 'Regiment' silhouette image from Carl.  Until then a temporary placement has been done.
image

Results
image

### CODE REVIEW

File: src/RollOfHonour.Web/Pages/Search/Results.razor [hunk 1]
```diff
@@ -224,37 +223,49 @@
...
+                break;
             }
 
-            MemorialSearchResults = results.Value;
+            case QueryType.Regiment:
+                await QueryPeopleByRegiment(false, PageIndex);
+                _regimentFilters = await _searchService.GetRegimentFiltersForSearchByRegimentName(BuildRegimentPersonQuery());
+                break;
+
+            default:
```
Comment (bashout) [inline src/RollOfHonour.Web/Pages/Search/Results.razor]: Should we redirect to a 404/500 error page? Throwing an exception here doesn't feel very friendly

File: src/RollOfHonour.Web/Pages/Search/Results.razor [hunk 2]
```diff
@@ -292,12 +303,52 @@
...
+            case QueryType.Regiment:
+                await QueryPeopleByRegiment(true, PageIndex);
+                break;
+            case QueryType.Person:
+                await QueryPeople(true, PageIndex);
+                break;
+            case QueryType.Memorial:
+                break;
+            default:
+                throw new ArgumentOutOfRangeException();
```
Comment (bashout) [inline src/RollOfHonour.Web/Pages/Search/Results.razor]: Same as the other point

**Review by bashout — CHANGES_REQUESTED** [review]

File: src/RollOfHonour.Web/Pages/Search/Results.razor (hunk 2 above)
Comment (DavidColemanUK) [reply src/RollOfHonour.Web/Pages/Search/Results.razor]: Good spot, arguably it's redundant code, so I've removed it.

File: src/RollOfHonour.Web/Pages/Search/Results.razor (hunk 1 above)
Comment (DavidColemanUK) [reply src/RollOfHonour.Web/Pages/Search/Results.razor]: Good spot, arguably it's redundant code, so I've removed it.

Comment (bashout) [reply src/RollOfHonour.Web/Pages/Search/Results.razor]: Is it redundant? What if someone links to a search with a bad URL? What gets returned?

Comment (DavidColemanUK) [reply src/RollOfHonour.Web/Pages/Search/Results.razor]: In that situation it shows like this, which I thin kit's pretty good.
image

**Review by bashout — APPROVED** [review]

**Pull request merged**